File: app/src/main/java/org/tasks/tasklist/ItemCallback.java
package org.tasks.tasklist;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import org.tasks.data.TaskContainer;

public class ItemCallback extends DiffUtil.ItemCallback<TaskContainer> {

  @Override
  public boolean areItemsTheSame(@NonNull TaskContainer oldItem, @NonNull TaskContainer newItem) {
    return oldItem.getId() == newItem.getId();
  }

  @Override
  public boolean areContentsTheSame(
      @NonNull TaskContainer oldItem, @NonNull TaskContainer newItem) {
    return oldItem.equals(newItem);
  }
}

File: app/src/main/java/org/tasks/tasklist/ItemCallback.kt
package org.tasks.tasklist

import androidx.recyclerview.widget.DiffUtil
import org.tasks.data.TaskContainer

internal class ItemCallback : DiffUtil.ItemCallback<TaskContainer>() {
    override fun areItemsTheSame(old: TaskContainer, new: TaskContainer) = old.id == new.id

    override fun areContentsTheSame(old: TaskContainer, new: TaskContainer) = old == new
}